lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -1540,15 +1540,28 @@ namespace Js
                 scriptContext->GetOrAddPropertyRecord(propertyName->GetString(), propertyName->GetLength(), &propertyRecord);
                 nextKey = propertyRecord->GetPropertyId();
             }
+            PropertyString * propertyString = PropertyString::TryFromVar(propertyName);
 
-            if (!JavascriptOperators::GetOwnProperty(from, nextKey, &propValue, scriptContext))
+            PropertyValueInfo getPropertyInfo;
+            if (!propertyString || !propertyString->TryGetPropertyFromCache(from, from, &propValue, scriptContext, &getPropertyInfo))
```

Comment:
this set of changes seem to be the core of the perf improvement?
please add a brief comment for how/why this is a win. (or, update the comment at the top of the while loop?)